test/qlever/test_stop_execute.py

```diff
@@ -0,0 +1,226 @@
+from __future__ import annotations
+import unittest
+from unittest.mock import patch, MagicMock
+from qlever.commands.stop import StopCommand
+
+
+class TestStopCommand(unittest.TestCase):
+
+    @patch('qlever.commands.stop.StatusCommand.execute')
+    @patch('psutil.process_iter')
+    @patch('qlever.containerize.Containerize.stop_and_remove_container')
+    @patch('qlever.commands.stop.StopCommand.show')
+    def test_execute_no_matching_processes_or_containers(self, mock_show,
+                                                mock_stop_and_remove_container,
+                                                mock_process_iter,
+                                                mock_status_execute):
+        # Setup args
+        args = MagicMock()
+        args.cmdline_regex = "ServerMain.* -i [^ ]*%%NAME%%"
+        args.name = "TestName"
+        args.no_containers = True

```

Comment:
I think (for this whole file) we agreed to split up the `execute` function, to make it easier testable
(separate functions for long `if` or `try` statements etc.
I am sure, if you do this , you can simplify the tests (first change the implementations, tests should still work, then simplify the tests).